sklearn/ensemble/_gb.py

```diff
@@ -23,33 +23,246 @@
 from abc import ABCMeta
 from abc import abstractmethod
 from numbers import Integral, Real
+from time import time
 import warnings
 
-from ._base import BaseEnsemble
-from ..base import ClassifierMixin, RegressorMixin
-from ..base import is_classifier
-
-from ._gradient_boosting import predict_stages
-from ._gradient_boosting import predict_stage
-from ._gradient_boosting import _random_sample_mask
-
 import numpy as np
-
-from scipy.sparse import csc_matrix
 from scipy.sparse import csr_matrix
 from scipy.sparse import issparse
 
-from time import time
 from ..model_selection import train_test_split
 from ..tree import DecisionTreeRegressor
 from ..tree._tree import DTYPE, DOUBLE
-from . import _gb_losses
 
-from ..utils import check_array, check_random_state, column_or_1d
+from .._loss.loss import (
+    _LOSSES,
+    AbsoluteError,
+    BaseLoss,
+    ExponentialLoss,
+    HalfBinomialLoss,
+    HalfMultinomialLoss,
+    HalfSquaredError,
+    HuberLoss,
+    PinballLoss,
+)
+from ..base import ClassifierMixin
+from ..base import RegressorMixin
+from ..base import is_classifier
+from ..dummy import DummyRegressor, DummyClassifier
+from ..exceptions import NotFittedError
+from ..preprocessing import LabelEncoder
+from ..tree._tree import TREE_LEAF
+from ..utils import check_random_state
+from ..utils import check_array
+from ..utils import column_or_1d
 from ..utils._param_validation import HasMethods, Interval, StrOptions
+from ..utils.fixes import percentile
 from ..utils.validation import check_is_fitted, _check_sample_weight
 from ..utils.multiclass import check_classification_targets
-from ..exceptions import NotFittedError
+from ..utils.stats import _weighted_percentile
+
+from ._base import BaseEnsemble
+from ._gradient_boosting import predict_stages
+from ._gradient_boosting import predict_stage
+from ._gradient_boosting import _random_sample_mask
+
+
+_LOSSES = _LOSSES.copy()
+_LOSSES.update(
+    {
+        "quantile": PinballLoss,
+        "huber": HuberLoss,
+    }
+)
+
+
+def _init_raw_predictions(X, estimator, loss, use_predict_proba):
+    """Return the initial raw predictions.
+
+    Parameters
+    ----------
+    X : ndarray of shape (n_samples, n_features)
+        The data array.
+    estimator : object
+        The estimator to use to compute the predictions.
+    loss : BaseLoss
+        An instace of a loss function class.
+    use_predict_proba : bool
+        Whether estimator.predict_proba is used instead of estimator.predict.
+
+    Returns
+    -------
+    raw_predictions : ndarray of shape (n_samples, K)
+        The initial raw predictions. K is equal to 1 for binary
+        classification and regression, and equal to the number of classes
+        for multiclass classification. ``raw_predictions`` is casted
+        into float64.
+    """
+    # TODO: Use loss.fit_intercept_only where appropriate instead of
+    # DummyRegressor.
```

Comment:
I can see that `estimator` is `Dummy*`, but it is hard to tell without knowing how `_init_raw_predictions` is used. Can this comment be expanded with this context?

Reply:
Better with https://github.com/scikit-learn/scikit-learn/pull/26278/commits/0c1c552bac8b695c399fcd270dfb8e904d3b5b8e?